### ISSUE TITLE
Enhancement: allow the tray with pager be floating and autohiding

### ISSUE THREAD

**Issue by dumblob** [joewing/jwm#346]
Currently the tray can't be arbitrarily positioned, but only always touch at least one screen corner. I would like to have a tray arbitrariy positioned - so that the tray center (i.e. the cross of the tray diagonals) could be positioned at 70% of the screen height and at 50% of the screen width.

Such tray would be then activated with a keyboard shortcut or by switching virtual desktops or any other event. The tray would then automatically hide itself after a user-set countdown timer expired (the timer was triggered by a last click somewhere in the tray).

**Comment by joewing** [comment]
Trays can be arbitrarily positioned right now (the `x`, `y`, `width`, and `height` components on the [`Tray`](http://joewing.net/projects/jwm/config.shtml#trays) tag).  Auto-hiding is a little strange for such trays, but it there is a key binding to show them (`showtray`).  It might not give you exactly what you want, but hopefully it gets you closer.

**Comment by dumblob** [comment]
Oh, I'm sorry for not noticing trays can be arbitrarily positioned (while cleaning my stash, I just put out old stuff without validity checking). After closely examining what can be done, I'd like to ask for the following features.

1. Extend the `Tray` tag attribute `autohide` with the value `"on"` (as opposed to the existing `"off"`) with the meaning, that there shouldn't be any one-pixel line at either screen side (and thus the only possibility to show the tray would be some other trigger).

1. Add the attribute `autohidedelay="500"` (500 is a non-negative integer number designating miliseconds) to the `Tray` tag to simply delay auto hiding.

1. Introduce a new global tag `Trigger` with the attribute `on="desktopchange"` (and possibly other values in the future) supporting the very same set of actions as the `Key` tag supports (at least `showtray`).

1. For tags `Tray` and `Pager` (and possibly also other tags allowed inside the `Tray` tag) add the attribute `ignoreinput="mouse"` (allowed values `mouse` and `keyboard`). If `ignoreinput` is specified more than once, it should ignore all the specified inputs.

1. Differentiate Trays in the keyboard shortcut `showtray`, e.g. by using the same double colon syntax as already used at some *jwmrc* places, namely `showtray:2` (of course, number the trays accordingly - the question is if the numbering should be implicit/internal or explicitly expected somewhere in the `Tray` tag).

**Comment by klausenbusk** [comment]
> Extend the Tray tag attribute autohide with the value "on" (as opposed to the existing "off") with the meaning, that there shouldn't be any one-pixel line at either screen side (and thus the only possibility to show the tray would be some other trigger).

I also need that feature for my use-case, any update on this?

**Comment by joewing** [comment]
Adding (1) `autohide="on"` to have the tray just disappear when not shown should be fairly easy.  Also, adding a delay for autohiding (2) should be easy.

I like the idea of adding a `Trigger` type binding (3).  I think that makes sense, especially with the mouse binding stuff I've been playing with recently (see #386).

As far as ignoring input is concerned (4), currently JWM doesn't do anything special with the keyboard for any tray components.  Obviously, mouse input is special on trays, though I'm not sure there's really a reason to ever ignore mouse input?

Adding a way to differentiate trays would be nice.  Currently this isn't done at all, so I'd have to think about how to do it (assign IDs explicitly perhaps?).

To summarize, I think (1) and (2) can/will happen pretty soon, along with maybe (3).  I'm currently trying to get 2.3.7 done and the next release will probably be 2.4.0, so that's where I see this happening.  (5) shouldn't be too difficult, either.  I'm not sure I understand what the use case is for (4).

**Comment by dumblob** [comment]
@joewing thank you for the positive feedback regarding (1) (2) (3) (5).

>(4) Obviously, mouse input is special on trays, though I'm not sure there's really a reason to ever ignore mouse input?

It boils down to two things. First, there are/will_be some default bindings (mouse & key) for trays and these bindings can't be easily reliably disabled. Second, there might be some issues with e.g. autohiding if any mouse pointer will be upon any widget in the tray or just the tray itself.

**Comment by dumblob** [comment]
@joewing it seems you're still quite busy, but I'm rude and will ask anyway :wink:.

Will you have some time to implement the above functionality?

**Comment by joewing** [comment]
Unfortunately, I probably won't get around to this any time soon.

**Comment by dumblob** [comment]
I have some new potential use cases for JWM. But would need this whole functionality. Anyone having time to pick up the gauntlet?
